.github/workflows/ci.yml

```diff
@@ -17,6 +17,39 @@ env:
   CACHE_NUMBER: 1
 
 jobs:
+  unit-test-frontend:
+    name: Frontend linting / unit tests
+    # Defines the type of runner the job runs on
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout to the repository
+        uses: actions/checkout@v3
+      - name: Set up NodeJS environment
+        uses: actions/setup-node@v3
+        with:
+          node-version: '14'
+      # Consider this as an add on to optimize the execution of actions
+      - name: Cache node modules
+        id: cache-npm
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-node-modules
+        with:
+          # npm cache files are stored in `~/.npm` on Linux/macOS
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+      - name: Install package dependencies
+        run: cd frontend2 && npm install
+      - name: Check linting and formatting
+        # Custom script for checking the linting and formatting being in place
+        run: cd frontend2 && npm run lint
```

Comment:
See the `working-directory` key in the example [here](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsrun).